homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,269 @@
+            self.update_supported_features(entity_id, None, new_state)
+        async_track_state_change(self.hass, self.entities,
+                                 state_change_listener)
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def assumed_state(self):
+        """Enable buttons even if at end position."""
+        return True
+
+    @property
+    def should_poll(self):
+        """Disable polling for the MultiCover."""
+        return False
+
+    @property
+    def is_closed(self):
+        """Return if all covers in group are closed."""
+        if not self.covers[KEY_OPEN_CLOSE]:
+            return False
+        for entity_id in self.covers[KEY_OPEN_CLOSE]:
+            state = self.hass.states.get(entity_id)
+            if state is None or state.state != STATE_CLOSED:
+                return False
+        return True
+
+    @property
+    def current_cover_position(self):
+        """Return current position for all covers.
+
+        None is unknown. pos if all have the same position.
+        Else 0 if closed or 100 is fully open.
+        """
+        position = -1
+        for entity_id in self.covers[KEY_POSITION]:
+            state = self.hass.states.get(entity_id)
+            pos = state.attributes.get(ATTR_CURRENT_POSITION)
+            if position == -1:
+                position = pos
+            elif position != pos:
+                position = -1
+                break
+
+        if position != -1:
+            return position
+
+        return 0 if self.is_closed else 100
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current tilt position for all covers.
+
+        None is unknown. pos if all have the same tilt position.
+        Else 100 for tilt open.
+        """
+        if self._tilt is False:
+            return None
+
+        position = -1
+        for entity_id in self.tilts[KEY_POSITION]:
+            state = self.hass.states.get(entity_id)
+            pos = state.attributes.get(ATTR_CURRENT_TILT_POSITION)
+            if position == -1:
+                position = pos
+            elif position != pos:
+                position = -1
+                break
+
+        if position != -1:
+            return position
+
+        return 100
+
+    @property
+    def state_attributes(self):
+        """Return the current position as attribute."""
+        data = {}
+        current_cover = self.current_cover_position
+        current_tilt = self.current_cover_tilt_position
+        if current_cover is not None:
+            data[ATTR_CURRENT_POSITION] = current_cover
+        if current_tilt is not None:
+            data[ATTR_CURRENT_TILT_POSITION] = current_tilt
+        return data
+
+    @property
+    def supported_features(self):
+        """Flag supported features for a cover."""
+        supported_features = 0
+
+        if self.covers[KEY_OPEN_CLOSE] != set():
```

Comment:
Same as before: just write `if not self.covers[KEY_OPEN_CLOSE]:` is faster by not requiring object creation